nixos/modules/system/boot/systemd.nix

```
@@ -226,7 +226,32 @@ in
     package = mkPackageOption pkgs "systemd" { };
 
     enableStrictShellChecks = mkEnableOption "" // {
-      description = "Whether to run shellcheck on the generated scripts for systemd units.";
+      description = ''
+        Whether to run `shellcheck` on the generated scripts for `systemd`
+        units.
+
+        When enabled, all systemd scripts generated by NixOS will be checked
+        with `shellcheck` and any errors or warnings will cause the build to
+        fail.
+
+        This affects all scripts that have been created through the `script`,
+        `reload`, `preStart`, `postStart`, `preStop` and `postStop` options for
+        `systemd` services. This does not affect command lines passed directly
+        to `ExecStart`, `ExecReload`, `ExecStartPre`, `ExecStartPost`,
+        `ExecStop` or `ExecStopPost`.
+
+        It therefore also does not affect `systemd` units that are coming from
```

```suggestion
        It therefore also does not affect systemd units that are coming from
```
And so is systemd, though it's stylised with a lowercase `s`.